lib/utils.ts

```diff
@@ -354,12 +354,10 @@ export function base32Encode(buffer: Buffer): string {
     return output;
 }
 
-export function splitArguments(options?: string): string[] {
-    return _.chain(
-        quoteParse(options || '')
-            // FIXME: x might not contain a .pattern!
-            .map((x: any) => (typeof x === 'string' ? x : (x.pattern as string))),
-    )
+export function splitArguments(options = ''): string[] {
+    // escape hashes first, otherwise they're interpreted as comments
+    const escapedOptions = options.replaceAll(/#/g, '\\#');
```

Comment:
C++ guy, not a JS guy, but is there any reason to be concerned that this could instead turn:
`\#` into `\\#`, and perhaps treat the `\` as `\` escaped, and still leave the `#` unescaped?

Reply:
That would be fine, considering then it would be considered a comment again and all the compiler arguments are removed.. I think

Reply:
I'll add an extra test to be sure before pushing this live. Thanks for the input

Reply:
Hrmph... I guess I don't really consider `\#` to typically mean 'an escaped # character'.  I see that `#` is a valid file name character on at least my linux box, and the `\` seems to be an optional escape character (That isn't necessary on my terminal, but IS ignored).

SO if you were to take a filename of : `clang some\#file#name.cpp` where `some#file#name.cpp` is a valid file name and only correct the 2nd hash, and not the 1st, that would likely be incorrect.

Additionally, I believe there is a way in compiler-explorer of including files on the command line from a URL, right?  What about if said URL has this combo of characters (as they are valid URL characters)?  

IMO, its a bit of a shame that 'comments' are supported by the library at all, they aren't a feature of my bash terminal (only when executed as a script), and I wouldn't expect CE to treat any amount of characters on the command line as a comment.

Reply:
1. So if you already had hashes in any argument, things would have been broken already. There's a reason we only got this bugreport now (after 10 years of CE), because noone sane would put # in their filename.
2. URL's should be urlencoded, otherwise it will only be known to the browser, this code applies to the backend only.
3. I agree that comments are a thing, but it also applies to things like > | < which have important meaning to the shell, but should absolutely not be parsed properly by our code. So in a way this module is quite helpful, just in case that shell commands with those characters get through to our servers.

I've added extra tests for what you proposed, and all looks fine https://github.com/compiler-explorer/compiler-explorer/commit/7573e69fbdbf216a6be3637c20c45fa51ca1e938

Reply:
As a compiler engineer, your Num1 there has made the biggest mistake there is, assuming users are sane :)  I also find bugs like this tend to happen in 3s, but *shrug*.

I'm generally glad that this fixes our warning flag, it was a shock when Aaron and I discovered it the other day.  So I'm OK with 'good enough', but wanted to make sure the edge-cases I'd thought of had been considered and dismissed intentionally.  

Thanks for the patch!

Reply:
Totally agree with your comments. But since we're mostly working in an execution jail for both compilations and user code execution, the only requirement really is not to be able to escape that jail.